### DISCUSSION TITLE
Databinding not working

### DISCUSSION THREAD

**Discussion by martinrhan** [AvaloniaUI/Avalonia#15621] [Q&A] [answered]
I have a class like this 
```
    public partial class Toolbox : INotifyPropertyChanged {
        public AvaloniaList<Tool> ToolList { get; } = new AvaloniaList<Tool>() {
            new Tool(){Name = "Tool 1"},
            new Tool(){Name = "Tool 2"}
        };

        [Notify] private Tool? selectedTool;
    }
```
Which is binded to a View
```
<UserControl>skiped codes here
  <TabControl ItemsSource="{Binding ToolList}" SelectedItem="{Binding SelectedTool}" TabStripPlacement="Left" VerticalContentAlignment="Top">
	<ItemsControl.ItemTemplate>
	  <DataTemplate>
		<TextBlock Text="{Binding Name}"></TextBlock>
	  </DataTemplate>
	</ItemsControl.ItemTemplate>
  </TabControl>
</UserControl>
````
The MainView is like this
```
<UserControl>skiped codes here
  <DockPanel>
	<v:ToolboxView DockPanel.Dock="Left" DataContext="{Binding Toolbox}"></v:ToolboxView>
	<v:WorldView DataContext="{Binding WorldViewModel}" SelectedTool="{Binding Path=Toolbox.SelectedTool, Mode=OneWay}"></v:WorldView>
  </DockPanel>
</UserControl>
```
However, when i check it with f12 dev tool, SelectedTool property in WorldView is null. I tried placing break point in the generated set method of SelectedTool, there was not null value set.
Here is a minimal reproduction of my bug
[AvaloniaApplication1.zip](https://github.com/AvaloniaUI/Avalonia/files/15218668/AvaloniaApplication1.zip)

**Comment by timunie** [comment, marked as answer]
does `WorldViewModel` have a property `Toolbox`?

**Reply by martinrhan** [reply]
How should I bind WorldView.SelectedTool with DataContext of MainView, while still setting DataContext of WorldView to WorldViewModel to let WorldView 's children bind to it?

**Reply by timunie** [reply]
https://docs.avaloniaui.net/docs/basics/data/data-binding/data-context
https://docs.avaloniaui.net/docs/basics/data/data-binding/compiled-bindings#type-casting
https://docs.avaloniaui.net/docs/guides/data-binding/binding-to-controls
